homeassistant/components/stream/core.py

```diff
@@ -356,3 +358,38 @@ async def handle(
     ) -> web.StreamResponse:
         """Handle the stream request."""
         raise NotImplementedError()
+
+
+class KeyFrameConverter:
+    """Generate and hold the keyframe as a jpeg."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+
+        # Keep import here so that we can import stream integration without installing reqs
+        # pylint: disable=import-outside-toplevel
+        from homeassistant.components.camera.img_util import TurboJPEGSingleton
```

Comment:
Is it a problem to make `camera` a `dependencies` for `stream`?  It seems like they would be loading it anyways

Reply:
There's also a circular import problem between the two components because camera also imports things from stream. The easiest way to avoid this is to rejig the imports (including the conditional ones) to avoid importing individual classes/functions from each other and instead importing the whole modules (ie avoid using `from x import y` and just using `import x`), but the former seems to be preferred in the HA codebase and the existing `stream` and `camera` code follow this. If we want to make those changes, I can do that in this PR or another one.